.github/workflows/build-and-test-differential.yaml

```
@@ -27,29 +27,58 @@ jobs:
     with:
       label: tag:require-cuda-build-and-test
 
-  build-and-test-differential:
+  prepare-build-and-test-differential:
+    runs-on: ubuntu-latest
     needs: [make-sure-label-is-present, make-sure-require-cuda-label-is-present]
-    if: ${{ needs.make-sure-label-is-present.outputs.result == 'true' }}
-    runs-on: ${{ matrix.runner }}
-    container: ${{ matrix.container }}${{ matrix.container-suffix }}
-    strategy:
-      fail-fast: false
-      matrix:
-        rosdistro:
-          - humble
-        container-suffix:
-          - ""
-          - -cuda
-        include:
-          - rosdistro: humble
-            container: ghcr.io/autowarefoundation/autoware:universe-devel
-            build-depends-repos: build_depends.repos
-          - container-suffix: -cuda
-            runner: codebuild-autoware-us-east-1-${{ github.run_id }}-${{ github.run_attempt }}-ubuntu-7.0-large
-            build-pre-command: taskset --cpu-list 0-5
-          - container-suffix: ""
-            runner: ubuntu-latest
-            build-pre-command: ""
+    outputs:
+      cuda_build: ${{ steps.check-if-cuda-build-is-required.outputs.cuda_build }}
+    steps:
+      - name: Check if cuda-build is required
+        id: check-if-cuda-build-is-required
+        run: |
+          if ${{ needs.make-sure-require-cuda-label-is-present.outputs.result == 'true' }}; then
+            echo "cuda-build is required"
+            echo "cuda_build=true" >> $GITHUB_OUTPUT
+          else
+            echo "cuda-build is not required"
+            echo "cuda_build=false" >> $GITHUB_OUTPUT
+          fi
+        shell: bash
+      - name: Fail if build is not required
```

```suggestion
      - name: Fail if the tag:run-build-and-test-differential is missing
```

This will convey the meaning better. Because, build is always required. But this will fail if the label is missing.